### PR TITLE
Opentiti fix

### PR DESCRIPTION
**Pull request by lvwerra**
Removes lines with html tags from text. Also fixes the pseudocrawl import.

### CODE REVIEW

File: clean_helpers/map_arabic.py [hunk 1]
```diff
@@ -2,4 +2,15 @@ def replace_newline_with_space(batch):
...
+def remove_html_spans(batch):
+    """Removes lines containing a '<' or '>' from the texts."""
+    bad_strings = ["<", ">"]
+    return {
+        **batch,
+        "text": [
+            "\n".join([line for line in text.split("\n") if not any([bs in line for bs in bad_strings])])
+            for text in batch["text"]
+        ]
+    }
```
Comment (thomasw21) [inline clean_helpers/map_arabic.py:16]: Hum that's just code duplication with the code one ^^. Ideally it'd be great to have a generic one. Though given the time constraint, maybe it's just better to duplicate code.

**Review by thomasw21 — APPROVED** [review]
So according to google sheet, we're going to use only `lm_ar_openiti_proc`:
 - I'm seeing some `PageV0MP004`
 - I'm not seeing any `<`, `>`
 - if we target html, shouldn't we be move specific list `</div>`, `<div>`, `<span>` etc ...
 
Otherwise looks good in terms of code.

File: clean_helpers/map_arabic.py (hunk 1 above)
Comment (lvwerra) [reply clean_helpers/map_arabic.py:16]: Yeah, I was a bit reluctant to add modify the existing one without testing it properly. I agree given more time we should just write a "remove html parts" function.

**Comment by lvwerra** [comment]
> So according to google sheet, we're going to use only `lm_ar_openiti_proc`:
> 
> * I'm seeing some `PageV0MP004`
> * I'm not seeing any `<`, `>`
> * if we target html, shouldn't we be move specific list `</div>`, `<div>`, `<span>` etc ...
> 
> Otherwise looks good in terms of code.

So I looked at lines in this dataset with both `<` and `>` (top 500 in terms of count). I double checked with @cakiki who said all of those can go. 

```
<div dir="rtl" id="book-container">
 <div>____________________
 < <
 > >
 </span>
 <span class="matn">. . . . . . . . . . . . . . . . . . . . . . . . . .  . . . . . . .
 </span><span class="matn-hr"> </span> [البناية]
 >
 </span><span class="matn-hr"> </span> [تصحيح الفروع للمرداوي]
 </span><span class="matn-hr"> </span> [فتح القدير]
 </span><span class="matn-hr"> </span> [منح الجليل]
 </span><span class="matn-hr"> </span> [العناية]
 </span><span class="matn-hr"> </span> [حاشية الشرواني]
 <span class="footnote">
 </span><span class="matn-hr"> </span> [رد المحتار]
 > > <
 </span></span>
 </span><span class="matn-hr"> </span> [حاشية الشبراملسي]
 </span><span class="matn-hr"> </span> [حاشية البجيرمي]
 </span><span class="matn-hr"> </span> [المبدع في شرح المقنع]
 </span><span class="matn-hr"> </span> [حاشية العدوي]
 </span><span class="matn-hr"> </span> [مغني المحتاج]
 </span><span class="matn-hr"> </span> [حاشية الجمل]
 </span><span class="matn-hr"> </span> [حاشية الصاوي]
 </span><span class="matn-hr"> </span> [نيل الأوطار]
 </span><span class="matn-hr"> </span> [منحة الخالق]
 > < <
 <
 </span><span class="matn-hr"> </span> [حاشية العبادي]
 </span><span class="matn-hr"> </span> [حاشية الدسوقي]
 </span><span class="matn-hr"> </span> [المنتقى]
 </span><span class="matn-hr"> </span> [طرح التثريب]
 </span><span class="matn-hr"> </span> [الشرح]
 </span><span class="matn-hr"> </span> [حاشية الشلبي]
 </span><span class="matn-hr"> </span> [حاشية الرملي الكبير]
 </span><span class="matn-hr"> </span> [حاشية قليوبي]
 </span><span class="matn-hr"> </span> [كشف الأسرار]
 </span><span class="matn-hr"> </span>. . . . . . . . . . . . . . . . .  . . . . . . . . . . . . . . . . .
 </span><span class="matn-hr"> </span> [غمز عيون البصائر]
 > <
 </span><span class="matn-hr"> </span> [سبل السلام]
 </span><span class="matn-hr"> </span> [حاشية العطار]
 </span><span class="matn-hr"> </span> [حاشية ابن الشاط = إدرار الشروق على أنواء الفروق]
 </span><span class="matn-hr"> </span> [تدريب الراوي]
 </span><span class="matn-hr"> </span> [حاشية الشرنبلالي]
 </span><span class="matn-hr"> </span> [التلويح]
 قوله عز وجل < <
 </span><span class="matn-hr"> </span> [الاختيار لتعليل المختار]
 </span><span class="matn-hr"> </span> [الفواكه الدواني]
 </span><span class="matn-hr"> </span> [العدة شرح العمدة]
 | < <
 </span></span></span>
 </span><span class="matn-hr"> </span>.
 </span>)
 </span><span class="matn-hr"> </span> [إحكام الأحكام]
 > بسم الله الرحمن الرحيم <
 > > 
 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .  .<span class="matn-hr"> </span> [تهذيب الفروق والقواعد السنية في الأسرار الفقهية]
 3]</span>
 5]</span>
 4]</span>
 11]</span>
 >الآثار الواردة في تفسير الآيات<
 10]</span>
 6]</span>
 <span class="red">
 7]</span>
 14]</span>
 8]</span>
 > | <
 16]</span>
 12]</span>
 1]</span>
 9]</span>
 26]</span>
 18]</span>
 35]</span>
 25]</span>
 22]</span>
 2]</span>
 17]</span>
 15]</span>
 32]</span>
 21]</span>
 20]</span>
 13]</span>
 28]</span>
 29]</span>
 40]</span>
 23]</span>
 > 
 44]</span>
 19]</span>
 30]</span>
 27]</span>
 24]</span>
 > } ) 71 <
 33]</span>
 37]</span>
 </span></span></span></span>
 31]</span>
 34]</span>
 / < <
 <div dir="rtl" id="book-container"> ا ل
 36]</span>
 50]</span>
 41]</span>
 >الآثار الواردة في تفسير الآيات وسبب النزول<
 39]</span>
 قوله عز وجل <
 48]</span>
 46]</span>
 60]</span>
 38]</span>
 42]</span>
 43]</span>
 <div dir="rtl" id="book-container">  ب
 49]</span>
 45]</span>
 70]</span>
 47]</span>
 <div dir="rtl" id="book-container"> و
 55]</span>
 > مقدمة <
 52]</span>
 61]</span>
 56]</span>
 62]</span>
 51]</span>
 <span class="matn">فصل
 57]</span>
 54]</span>
 59]</span>
 64]</span>
 > > 7 !
 53]</span>
 { < <
 >الآثار الوارده في تفسير الآيات وسبب النزول<
 58]</span>
 73]</span>
 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .  .<span class="matn-hr"> </span>
 67]</span>
 82]</span>
 74]</span>
top">الفهرست:
 </br></br></br></br>
 > فصل
 68]</span>
 </span>]
 71]</span>
 > .
 76]</span>
 > بسم الله الرحمن الرحيم < <
 2 < <
 3 < <
 1 < <
 > > وفيه مسائل :
 > بسم الله الرحمن الرحيم
 4 < <
 66]</span>
 5 < <
 65]</span>
 83]</span>
 72]</span>
 6 < <
 8 < <
 79]</span>
 <span class="footnote"> فتاوى المنار
 7 < <
top">السراج الوهاج:
 80]</span>
 87]</span>
 </br></br></br>
 77]</span>
 93]</span>
 >حول السورة<
 89]</span>
 </span></span></span></span></span>
 <span class="matn">000000000000000000000000000000000
 * * * <div>____________________
 63]</span>
 9 < <
 10 < <
 11 < <
 </span></span>)
 <div dir="rtl" id="book-container"> باب ا ل
 75]</span>
 91]</span>
 81]</span>
 12 < <
 78]</span>
 13 < <
 98]</span>
 @QB@ < <
 14 < <
 <span class="footnote"> الأخبار والآراء
 15 < <
 101]</span>
 <div dir="rtl" id="book-container">  و
 > @QB@ بسم الله الرحمن الرحيم @QE@
 16 < <
 17 < <
 69]</span>
 18 < <
 <div dir="rtl" id="book-container">  ف
 92]</span>
 19 < <
 104]</span>
 110]</span>
 84]</span>
 > > في الآية مسائل :
 86]</span>
 100]</span>
 20 < <
 </br></br></br></br></br>
 21 < <
 90]</span>
top">روضة  الطالبين ٣:
 22 < <
 95]</span>
 96]</span>
 >الآثار الوارده في تفسير الآيات<
top">المغني ١:
top">روضة الطالبين ٣:
 1 - < <
 23 < <
 24 < <
 88]</span>
 108]</span>
 26 < <
 قوله تعالى < <
 25 < <
 102]</span>
 103]</span>
 2 - < <
 27 < <
 99]</span>
 28 < <
 29 < <
 94]</span>
 113]</span>
 105]</span>
 115]</span>
 العين</span>
 30 < <
 </span>:
 > مكية
 > فصل <
 97]</span>
 </br></br>
 <div dir="rtl" id="book-container"> ا  ل
 3 - < <
 31 < <
top">الاستبصار ٢:
 109]</span>
 111]</span>
 > > فيه مسائل :
 4 - < <
 33 < <
 34 < <
 32 < <
 35 < <
 85]</span>
 د]</span> فعل، بفتح الفاء وسكون العين
 36 < <
 5 - < <
 37 < <
 114]</span>
 <span class="matn">فصل:
 38 < <
 </span><span class="matn-hr"> </span> تخريجه:
 <div dir="rtl" id="book-container">  ل
 <div dir="rtl" id="book-container"> الأ ف
 <span class="matn">فيه مسائل:
 39 < <
 40 < <
 117]</span>
 122]</span>
 <span class="footnote"> أميل القرن التاسع عشر
 <div dir="rtl" id="book-container">  م
 41 < <
 42 < <
top">الاستبصار ١:
<div dir="rtl" id="book-container">
 106]</span>
 <div class="quran">
 43 < <
 44 < <
 107]</span>
 112]</span>
 </span> المفتي
 6 - < <
 7 - < <
 45 < <
 (٢) <a href="[http://qadatona.org/عربي/القرآن-الكريم/5/6](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/5/6)"  target="_blank">المائدة: ٦</a>.
 <span class="footnote"> تقريظ المطبوعات الجديدة
top">رجال النجاشي:
 46 < <
 > > [ الآية : 1 ] .
 159]</span>
 47 < <
 48 < <
 49 < <
top">الخصال:
 8 - < <
 50 < <
 119]</span>
top">المجموع ٩:
 9 - < <
 (قوله:<span class="matn-hr"> </span> [حاشية ابن قاسم العبادي]
 51 < <
 52 < <
 > > [ الآية : 3 ] .
top">المقنع:
 </span> (1)
 <div dir="rtl" id="book-container"> الأفعال [ال م
 11 - < <
 </span><span class="matn-hr"> </span>فيه مسائل:
 <span class="matn">. . . . . . . . . . . . . . . . . . . . . . . . . .  . .
 <span class="red"> 
 129]</span>
 </span><span class="matn-hr"> </span>
 17 - < <
 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .  .<span class="matn-hr"> </span> [حاشية الشربيني]
top">النهاية:
 138]</span>
 </span></span></span></span></span></span>
 </br></br></br></br></br></br>
 > . 
 <div dir="rtl" id="book-container"> ف ع
 10 - < <
 53 < <
 <div dir="rtl" id="book-container"> 2 - 
 147]</span>
 ن</span>
 </br>
 54 < <
top">الاستبصار ٤:
 125]</span>
top">الاستبصار ٣:
 </span> [ا ل
 16 - < <
 * بسم الله الرحمن الرحيم <
 <div dir="rtl" id="book-container"> 1 - 
 <span class="matn">.. ... ... ... ... ... ... ... ... ... ... ... ...  ... ... ... ... ... ... ... ... ... ... ...
 <div dir="rtl" id="book-container">  ر
 13 - < <
 14 - < <
 55 < <
 > > .
 </span>1
 116]</span>
 <div dir="rtl" id="book-container"> الأفعال [المج ر
 18 - < <
 56 < <
 )</span>
 118]</span>
 152]</span>
 (٢) <a href="[http://qadatona.org/عربي/القرآن-الكريم/5/1](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/5/1)"  target="_blank">المائدة: ١</a>.
 136]</span>
 <div dir="rtl" id="book-container"> الأف ع
 الهاء</span>
 12 - < <
 57 < <
 58 < <
 59 < <
 151]</span>
top">المعارف:
 60 < <
 (٣) <a href="[http://qadatona.org/عربي/القرآن-الكريم/2/196](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/2/196)"  target="_blank">البقرة: ١٩٦</a>.
 (٣) <a href="[http://qadatona.org/عربي/القرآن-الكريم/5/1](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/5/1)"  target="_blank">المائدة: ١</a>.
 141]</span>
 > > وفيه مسألتان :
 (قوله<span class="matn-hr"> </span> [حاشية ابن قاسم العبادي]
 </span></span></span></span></span></span></span>
 132]</span>
 > | بسم الله الرحمن الرحيم { < <
 15 - < <
 61 < <
 62 < <
 130]</span>
 128]</span>
 137]</span>
 127]</span>
 63 < <
 64 < <
 (٣) <a href="[http://qadatona.org/عربي/القرآن-الكريم/5/6](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/5/6)"  target="_blank">المائدة: ٦</a>.
 143]</span>
 170]</span>
 154]</span>
 126]</span>
 </span>د]</span> فعل، بفتح الفاء وسكون العين
 الفاء</span>
top">صلاة الجمعة ص ١٠٣</a>.
 19 - < <
 24 - < <
 65 < <
 66 < <
 67 < <
 68 < <
 69 < <
 > > [ الآية : 7 ] .
 144]</span>
 164]</span>
 123]</span>
 140]</span>
 145]</span>
 148]</span>
top">الوسيلة:
top">الشرح الكبير ١٠:
 <div dir="rtl" id="book-container"> فع ل
 70 < <
 71 < <
 72 < <
 73 < <
 <div dir="rtl" id="book-container"> 4 - 
 <div dir="rtl" id="book-container"> 3 - 
 121]</span>
 120]</span>
 160]</span>
 131]</span>
 (٢) <a href="/الكتب/1167_التهذيب-ج-٧/الصفحة_0?pageno=371
top">مغني المحتاج ١:
top">المجموع ٧:
 74 < <
 75 < <
top">المناقب:
 </span> 1- 
 186]</span>
 (٢) <a href="[http://qadatona.org/عربي/القرآن-الكريم/2/196](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/2/196)"  target="_blank">البقرة: ١٩٦</a>.
 124]</span>
 135]</span>
 </span>2
 <div dir="rtl" id="book-container"> 6 - 
 <span class="footnote"> الأخبار التاريخية
 (الگلپايگاني).</div> 
 23 - < <
top">الفقيه ٢:
top">بصائر الدرجات:
 > > [ الآية : 5 ] .
 > > [ الآية : 2 ] .
 > > [ الآية : 10 ] .
 133]</span>
 179]</span>
 163]</span>
 153]</span>
top">إحقاق الحق:
 وقوله تعالى < <
top">المقنعة:  ٧٠</a>.
 <span class="matn">. . . . . . . . . . . . . . . . . . . . . . . . . .  . . . . .
 (٢) <a href="[http://qadatona.org/عربي/القرآن-الكريم/2/275](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/2/275)"  target="_blank">البقرة: ٢٧٥</a>.
top">سنن ابن ماجة  ١:
 > > وفي الآية مسائل :
 <div dir="rtl" id="book-container"> الأفعال [المج ر د
 20 - < <
 22 - < <
 76 < <
 77 < <
 > > [ الآية : 4 ] .
 > > [ الآية : 9 ] .
 > > [ الآية : 15 ] .
 146]</span>
 188]</span>
top">المغني ٤:
 <div dir="rtl" id="book-container"> ق
 <div dir="rtl" id="book-container">   ب
top">فتح القدير ١:
top">المغني ٣:
 28 - < <
 78 < <
top">التهذيب ٤:
 > > [ الآية : 19 ] .
top">التهذيب ٥:
 <span class="matn">(فصل)
top">فهرست منتجب الدين:
 (٢) <a href="[http://qadatona.org/عربي/القرآن-الكريم/5/3](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/5/3)"  target="_blank">المائدة: ٣</a>.
 (٤) <a href="[http://qadatona.org/عربي/القرآن-الكريم/4/11](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/4/11)"  target="_blank">النساء: ١١</a>.
top">الفقيه:
 > ( فصل ) <
top">سنن أبي داود  ١:
 ^ { فبأي آلاء ربكما تكذبان } < <
top">المجموع ٢:
 26 - < <
 </span><span class="matn-hr"> </span> [البناية ]
 > > [ الآية : 13 ] .
 > > [ الآية : 26 ] .
 > > [ الآية : 21 ] .
 </span> 3 - 
 155]</span>
 (٥) <a href="[http://qadatona.org/عربي/القرآن-الكريم/5/6](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/5/6)"  target="_blank">المائدة: ٦</a>.
 (٢) <a href="[http://qadatona.org/عربي/القرآن-الكريم/4/11](http://qadatona.org/%D8%B9%D8%B1%D8%A8%D9%8A/%D8%A7%D9%84%D9%82%D8%B1%D8%A2%D9%86-%D8%A7%D9%84%D9%83%D8%B1%D9%8A%D9%85/4/11)"  target="_blank">النساء: ١١</a>.
 > > @QB@ فبأي آلاء ربكما تكذبان > @QB@ <
 </span>3
top">سنن ابن ماجة ١:
 ]</span> فعل،  ب
 <div dir="rtl" id="book-container"> باب ا  ل
 عال</span> [ال م
 <div dir="rtl" id="book-container">  ي
 <div dir="rtl" id="book-container"> و [ ف
 <span class="footnote"> التقريظ
 25 - < <
 29 - < <
```

File: clean_helpers/map_arabic.py (hunk 1 above)
Comment (thomasw21) [reply clean_helpers/map_arabic.py:16]: No I mean
```
def buid_bad_substring_remover(bs: List[str]):
   def remove_bad_substring(batch):
        return {
            **batch,
            "text": [
                "\n".join([line for line in text.split("\n") if not any([bs in line for bs in bad_strings])])
                for text in batch["text"]
            ]
        }
    return remove_bad_substring
```

Comment (lvwerra) [reply clean_helpers/map_arabic.py:16]: oh yeah, that's a good idea, let me do that.

**Comment by lvwerra** [comment]
@thomasw21 I generalized the function and made the removal a bit more specific. Wdyt?

File: clean.py [hunk 2]
```diff
@@ -17,7 +17,8 @@
 # Map functions: function(batch: Dict) -> Dict
 MAPS = {
     "replace_newline_with_space": replace_newline_with_space,
-    "remove_lines_with_code": remove_lines_with_code
+    "remove_lines_with_code": build_bad_substring_remover(["{", "}", "[if", "<script"]), 
+    "remove_html_spans": build_bad_substring_remover(["<span", "<div", "<a", "/span>", "br>"])
```
Comment (thomasw21) [inline clean.py:21]:
```suggestion
    "remove_html_spans": build_bad_substring_remover(["<span", "</span>", "<div", "</div>", "<a", "</a>", "br>"])
```

**Review by thomasw21 — APPROVED** [review]
Factorisation looks good to me! Thank you!

**Pull request merged**